File: Python/643.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        max_ = sum(nums[:k])
        total = sum(nums[:k])
        for i in range(len(nums)-k):
            total = total + nums[i+k] - nums[i]
            if total > max_:
                max_ = total
        return float(max_)/k

File: Java/643.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int sum = 0, cur = 0;
        for(int i = 0; i < k; i++) sum += nums[i];
        cur = sum;
        for(int i = 0; i < nums.length-k; i++){
            cur = cur - nums[i] + nums[i+k];
            if(cur > sum) sum = cur;
        }
        return Double.valueOf(sum)/k;
    }
}